src/basic/build-path.c

```diff
@@ -277,7 +277,9 @@ int pin_callout_binary(const char *path, char **ret_path) {
                         return fd;
         }
 
-        r = open_and_check_executable(path, /* root = */ NULL, ret_path, &fd);
+        r = find_executable_full(path, /* root = */ NULL,
```

Comment:
Please drop the last commit. Currently, this function is only used by the service manager and udevd, both does not support `$PATH`.

Or, maybe better to make `pin_callout_binary()` optionally take `exec_search_path`. Then we can drop the following lines:
https://github.com/systemd/systemd/blob/72008278857fdcf5822fbdb903105d293c4251ff/src/udev/udev-spawn.c#L273-L282

Reply:
I don't see the need of dropping it? Both pid1 and udev enforce absolute path before calling pin_callout_binary(), so this has no change of any behavior? The commit merely brings invoke_ and pin_ variants on par.

Reply:
Sorry, yeah, dropping the commit is overclaim.